.changeset/long-brooms-shake.md

```diff
@@ -0,0 +1,5 @@
+---
+"react-router": patch
+---
+
+[UNSTABLE] Internal refactor to how `unstable_onError` is called by `RouterProvider`
```

Comment:
```suggestion
[UNSTABLE] Add `unstable_pattern` to the parameters for client side `unstable_onError`, refactor how it's called by `RouterProvider` to avoid potential strict mode issues
```